client/notifications/notification-reducer.ts

```diff
@@ -57,6 +57,11 @@ function updateReadStatus(
   )
 }
 
+// Cache the values of the cleared notifications so they can easily be restored if the server
+// returns an error.
+let clearedIdToNotification: Map<string, NotificationRecord>
```

Comment:
This makes the reducer non-pure, bad idea

If you want to store these, they should be included in the reducer's state